File: lib/core/services/auth_service.dart
// Firebase Auth Service cho Flutter
// Cung cấp các phương thức xác thực người dùng

import 'package:firebase_auth/firebase_auth.dart';

class AuthService {
  static final FirebaseAuth _auth = FirebaseAuth.instance;

  // Lấy user hiện tại
  static User? get currentUser => _auth.currentUser;

  // Stream theo dõi trạng thái Auth (dùng trong StreamProvider/StreamBuilder)
  static Stream<User?> get authStateChanges => _auth.authStateChanges();

  // ========================
  // ĐĂNG KÝ bằng Email/Password
  // ========================
  static Future<UserCredential> signUp({
    required String email,
    required String password,
    String? displayName,
  }) async {
    final credential = await _auth.createUserWithEmailAndPassword(
      email: email,
      password: password,
    );
    if (displayName != null) {
      await credential.user?.updateDisplayName(displayName);
    }
    return credential;
  }

  // ========================
  // ĐĂNG NHẬP bằng Email/Password
  // ========================
  static Future<UserCredential> signIn({
    required String email,
    required String password,
  }) async {
    return await _auth.signInWithEmailAndPassword(
      email: email,
      password: password,
    );
  }

  // ========================
  // ĐĂNG XUẤT
  // ========================
  static Future<void> signOut() async {
    await _auth.signOut();
  }

  // ========================
  // QUÊN MẬT KHẨU
  // ========================
  static Future<void> sendPasswordResetEmail(String email) async {
    await _auth.sendPasswordResetEmail(email: email);
  }

  // ========================
  // CẬP NHẬT thông tin profile
  // ========================
  static Future<void> updateProfile({
    String? displayName,
    String? photoURL,
  }) async {
    await _auth.currentUser?.updateDisplayName(displayName);
    await _auth.currentUser?.updatePhotoURL(photoURL);
  }
}

File: src/utils/authService.js
import { 
  createUserWithEmailAndPassword, 
  signInWithEmailAndPassword,
  updateProfile,
  signOut
} from 'firebase/auth';
import { auth } from '../config/firebase';

/**
 * Đăng nhập
 * Chỉ thực hiện xác thực với Firebase.
 * Việc đồng bộ sẽ được xử lý bởi SyncLoadingScreen sau khi điều hướng.
 */
export const loginUser = async (email, password) => {
  const userCredential = await signInWithEmailAndPassword(auth, email, password);
  return userCredential.user;
};

/**
 * Đăng ký
 * Tạo tài khoản Firebase và cập nhật thông tin cơ bản.
 */
export const registerUser = async (name, email, password) => {
  const userCredential = await createUserWithEmailAndPassword(auth, email, password);
  
  // Cập nhật tên hiển thị ngay lập tức
  await updateProfile(userCredential.user, { displayName: name });
  
  return userCredential.user;
};

/**
 * Đăng xuất
 */
export const logoutUser = async () => {
  await signOut(auth);
};
